### PR TITLE
implement webhooks CRUD

### PR DESCRIPTION
**Pull request by Gerdie**
Adds create, read, update and delete functionality for webhooks.

### CODE REVIEW

File: src/nylas.js [hunk 1]
```diff
@@ -29,16 +30,15 @@ class Nylas {
       this.apiServer = apiServer;
     }
 
-    let conn;
+    const conn = new NylasConnection(this.appSecret, { clientId: this.appId });
+    this.webhooks = new ManagementModelCollection(Webhook, conn, this.appId);
```
Comment (smrutherford) [inline src/nylas.js]: oooh smart - i like how you made it accessible on the Nylas model so ppl don't have to make their own instance of an account management model

File: src/models/restful-model.js [hunk 2]
```diff
@@ -46,6 +46,15 @@ export default class RestfulModel {
     return json;
   }
 
+  // Subclasses should override this method.
+  pathPrefix() {
+    return '';
+  }
+
+  saveEndpoint() {
+    return `${this.pathPrefix()}/${this.constructor.collectionName}`;
```
Comment (smrutherford) [inline src/models/restful-model.js]: crafty! 💯

File: src/models/restful-model-collection.js [hunk 3]
```diff
@@ -208,7 +208,7 @@ export default class RestfulModelCollection {
   }
 
   build(args) {
-    const model = new this.modelClass(this.connection);
+    const model = this._createModel({});
```
Comment (smrutherford) [inline src/models/restful-model-collection.js]: 👍

File: README.md [hunk 4]
```diff
@@ -371,6 +371,30 @@ nylas.messages.search('Hey!').then(messages => console.log(messages));
...
+```javascript
+// create a new webhook
+const newWebhook = Nylas.webhooks.build({
+  callbackUrl: 'https://wwww.myapp.com/webhook',
+  state: 'active',
+  triggers: ['event.created', 'event.updated'],
+  });
+newWebhook.save().then(webhook => console.log(webhook.id));
+
+// get and update an existing webhook
```
Comment (yusra-ahmed) [inline README.md]: the readme might also be a good place to explain that you can only update state, if they want to update anything else they'll have to make a new post.

**Pull request merged**